packages/server/lib/controllers/config.controller.ts

```diff
@@ -355,22 +366,38 @@ class ConfigController {
                 const availableFlows = flowService.getAllAvailableFlowsAsStandardConfig();
                 const [availableFlowsForProvider] = availableFlows.filter((flow) => flow.providerConfigKey === config.provider);
 
-                const enabledFlows = await getConfigWithEndpointsByProviderConfigKey(environmentId, providerConfigKey);
-                const unEnabledFlows: StandardNangoConfig = availableFlowsForProvider as StandardNangoConfig;
-
-                if (availableFlows && enabledFlows && unEnabledFlows) {
-                    const { syncs: enabledSyncs, actions: enabledActions } = enabledFlows;
-
-                    const { syncs, actions } = unEnabledFlows;
-
-                    const filteredSyncs = syncs.filter((sync) => !enabledSyncs.some((enabledSync) => enabledSync.name === sync.name));
-                    const filteredActions = actions.filter((action) => !enabledActions.some((enabledAction) => enabledAction.name === action.name));
-
-                    unEnabledFlows.syncs = filteredSyncs;
-                    unEnabledFlows.actions = filteredActions;
+                const enabledAndDisabledFlows = await getConfigWithEndpointsByProviderConfigKey(environmentId, providerConfigKey);
+                const publicFlows: StandardNangoConfig = availableFlowsForProvider as StandardNangoConfig;
+                const disabledFlows = publicFlows;
+
+                if (availableFlows && enabledAndDisabledFlows && publicFlows) {
+                    const { syncs, actions } = enabledAndDisabledFlows;
+                    const { enabledSyncs, disabledSyncs } = syncs.reduce<SyncConfigs>(
+                        (acc, sync: NangoSyncConfig) => (sync.enabled ? acc.enabledSyncs.push(sync) : acc.disabledSyncs.push(sync), acc),
+                        { enabledSyncs: [], disabledSyncs: [] }
+                    );
+                    const { enabledActions, disabledActions } = actions.reduce<ActionConfigs>(
+                        (acc, action: NangoSyncConfig) => (action.enabled ? acc.enabledActions.push(action) : acc.disabledActions.push(action), acc),
+                        { enabledActions: [], disabledActions: [] }
+                    );
+
+                    enabledAndDisabledFlows.syncs = enabledSyncs;
+                    enabledAndDisabledFlows.actions = enabledActions;
```

Comment:
`enabledAndDisabledFlows` is confusing: at first it contains all flows but here we mutate it to only have enabled syncs/actions